packages/runtime/src/state/actions.ts

```diff
@@ -413,18 +466,20 @@ export function registerComponentEffect(
 }
 
 export function registerBuilderComponent(
-  type: string,
-  meta: ComponentMeta,
-  serializedControls: Record<string, SerializedControl>,
+  payload: RegisterBuilderComponentAction['payload'],
+  transferables?: Transferable[],
 ): RegisterBuilderComponentAction {
   return {
     type: ActionTypes.REGISTER_BUILDER_COMPONENT,
-    payload: { type, meta, serializedControls },
+    payload,
+    transferables,
   }
 }
 
-export function unregisterBuilderComponent(type: string): UnregisterBuilderComponentAction {
-  return { type: ActionTypes.UNREGISTER_BUILDER_COMPONENT, payload: { type } }
+export function unregisterBuilderComponent(
+  payload: UnregisterBuilderComponentAction['payload'],
```

Comment:
Switched to taking the payload as named params, this caught a [long-lurking bu](https://github.com/makeswift/makeswift/pull/894/files#diff-366c80f0759766e0e928735657a47bda3f3b9beed34394c1768b11c04879b2b0L444)g.